src/tracer/MOM_generic_tracer.F90

```diff
@@ -582,7 +634,8 @@ subroutine MOM_generic_tracer_column_physics(h_old, h_new, ea, eb, fluxes, Hml,
       call generic_tracer_source(tv%T, tv%S, rho_dzt, dzt, dz_ml, G%isd, G%jsd, 1, dt, &
                G%areaT, get_diag_time_end(CS%diag), &
                optics%nbands, optics%max_wavelength_band, optics%sw_pen_band, optics%opacity_band, &
-               internal_heat=tv%internal_heat, frunoff=fluxes%frunoff, sosga=sosga, geolat=G%geolatT, eqn_of_state=tv%eqn_of_state)
+               internal_heat=tv%internal_heat, frunoff=fluxes%frunoff, sosga=sosga, geolat=G%geolatT, eqn_of_state=tv%eqn_of_state, &
```

Comment:
We can consider remove `eqn_of_state=tv%eqn_of_state`

Reply:
Agreed - once we have decided that these results are similar enough to those using the photoacclimation depth calculated in COBALT we can change this interface.

Reply:
I think once this PR and the ocean_bgc PR are merged, I will open two more which remove the extra arguments.